includes/Frontend/Frontend_Form.php

```
@@ -238,8 +247,30 @@ public function draft_post() {
+                }
+            } else {
+                // Guest users: validate draft token to prevent IDOR
+                $stored_token    = get_post_meta( $update_post_id, '_wpuf_draft_token', true );
+                $submitted_token = isset( $_POST['_wpuf_draft_token'] ) ? sanitize_text_field( wp_unslash( $_POST['_wpuf_draft_token'] ) ) : '';
+
+                if ( empty( $stored_token ) || ! hash_equals( $stored_token, $submitted_token ) ) {
+                    wp_send_json_error( [ 'message' => __( 'You are not authorized to edit this draft.', 'wp-user-frontend' ) ] );
+                }
+            }
+
             $is_update                 = true;
```

PHPMD (2.15.0)

[warning] Avoid unused local variables such as '$is_update'. (undefined)

(UnusedLocalVariable)